docs/dev.rst

```diff
@@ -1,20 +1,574 @@
+
+
+Set the Retrieve State
+^^^^^^^^^^^^^^^^^^^^^^
+
+To make changes to retrieve in an org, use the ``snapshot_changes`` task to create a snapshot. A snapshot tells the Salesforce CLI source tracking to set the org's current state as a baseline for changes to be made against.
+
+.. code-block:: console
+
+    $ cci task run snapshot_changes --org dev
+
+.. note:: There's no need to run ``snapshot_changes`` if you've set up a scratch org with any of these flows.
+
+    * ``dev_org``
+    * ``dev_org_namespaced``
+    * ``qa_org``
+    * ``regression_org``
+    * ``install_beta``
+    * ``install_prod``
+    
+    These flows all run ``snapshot_changes`` as their last step.
+
+The ``list_changes`` and ``retrieve_changes`` tasks detect new or modified metadata in an org, but ignore any changes to metadata made prior to the last snapshot.
+
+A snapshot is created successfully when the final line of output for ``list_changes`` and ``retrieve_changes`` reads "Found no changes."
+
+.. code-block:: console
+
+    $ cci task run list_changes --org dev
+
+    YYYY-MM-DD HH:MM:SS: Getting org info from Salesforce CLI for <username>
+    YYYY-MM-DD HH:MM:SS: Beginning task: ListChanges
+    YYYY-MM-DD HH:MM:SS: As user: <username>
+    YYYY-MM-DD HH:MM:SS: In org: <org_name>
+    YYYY-MM-DD HH:MM:SS:
+    YYYY-MM-DD HH:MM:SS: Found no changes.
+
+
+
+List Changes
+------------
+
+To see what components have changed in a target org:
+
+.. code-block:: console
+
+    $ cci task run list_changes --org dev
+
+.. note::
+    
+    This functionality relies on Salesforce's source tracking feature, which is currently available only in Scratch Orgs, Developer Sandboxes, and Developer Pro Sandboxes.
+
+For more information, see `List and Retrieve Options`_.
+
+Retrieve Changes
+----------------
+
+The ``retrieve_changes`` task supports both Salesforce DX- and Metadata API-format source code. It utilizes the `SourceMember <https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_sourcemember.htm>`_
+``sObject`` to detect what has changed in an org, and also gives you discretion regarding which components are retrieved when compared to the ``dx_pull`` task.
+
+Manual tracking of component versions also offers the possibility of retrieving changes into one directory, and then running the task again to retrieve other changes into a different directory.
+ 
+To retrieve all changes in an org:
+
+.. code-block:: console
+
+    $ cci task run retrieve_changes --org dev
+
+For more information, see `List and Retrieve Options`_.
+
+
+
+List and Retrieve Options 
+-------------------------
+
+When developing in an org, the changes you're most interested in are sometimes mixed with other changes that aren't relevant to what you're doing.
+
+    Example: Changing schema like ``Custom Objects`` and ``Custom Fields`` often results in changes to ``Page Layouts`` and ``Profiles`` that you don't wish to review or retrieve.
```

Comment:
Added backquotes to these objects. Is this correct?

Reply:
I think here we can omit them - we're referring to their user-facing labels rather than their underlying API names.